opentelemetry-otlp/src/metric.rs

```diff
@@ -100,6 +100,56 @@ impl From<TonicExporterBuilder> for MetricsExporterBuilder {
     }
 }
 
+/// Configure delta temporality for all [InstrumentKind]
+///
+/// [Temporality::Delta] will be used for Counter, Asynchronous Counter and Histogram
+/// [Temporality::Cumulative] will be used for UpDownCounter and Asynchronous UpDownCounter
+#[derive(Clone, Default, Debug)]
+struct DeltaTemporalitySelector {
+    pub(crate) _private: (),
+}
+
+impl DeltaTemporalitySelector {
+    /// Create a new default temporality selector.
+    fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl TemporalitySelector for DeltaTemporalitySelector {
+    fn temporality(&self, _kind: InstrumentKind) -> Temporality {
+        Temporality::Delta
+    }
+}
+
+/// Configure low memory temporality for all [InstrumentKind]
+///
+/// [Temporality::Delta] will be used for Counter and Histogram
+/// [Temporality::Cumulative] will be used for UpDownCounter, Asynchronous Counter and Asynchronous UpDownCounter
+#[derive(Clone, Default, Debug)]
+struct LowMemoryTemporalitySelector {
+    pub(crate) _private: (),
+}
+
+impl LowMemoryTemporalitySelector {
+    /// Create a new default temporality selector.
+    fn new() -> Self {
```

Comment:
nit: as the lint is pointing out, are you intending this to be `pub`? or should this be removed as it's not used.